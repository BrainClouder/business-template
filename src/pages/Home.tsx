import React from 'react';
import AppBar from '../_components/AppBar';
import styled from 'styled-components';

const isMobile = window.innerWidth < 600;
const FirstContainer = styled.div`
    position: relative;
    // background-color: hsl(221, 51%, 14%);
    // height: calc(100vh - 2.5em);
    width: 100vw;
    padding: 4.5em 0;
    z-index: 4;
`;

const SecondContainer = styled.div`
    width: 100vw;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color:  hsl(221, 51%,95%);

`;

const ThirdContainer = styled.div`

`;

const ExplainBox = styled.div`
    box-shadow: 2px 2px 2px #112;
    background-color: white;
    font-weight: 300;
    font-size: 16px;
    margin: 1em 0;
    text-align: center;
    padding: 1em 1.5em;
    width: 80vw;    
`;

const LightText = styled.p`
    font-size: 32px;
    font-weight: 300;
    padding: 2px 4px;
    color:  hsl(221, 51%, 84%);    
    text-align: center;
    z-index: 5;
`;

const Hl = styled.span`
    font-weight: 600;
`;

const BlueCircle = styled.div`
    position: absolute;
    top: 40vh;
    width: 60vw;
    left: 0vw;
    height: 150px;
    border-bottom: 1.5em solid hsl(221, 51%, 94%);
    // border-radius: 50%;
    border-bottom-right-radius: 100%;
    z-index: 0;
`;

const FirstPlanet = styled.div`
    width: 50px;
    height: 50px;
    border-radius: 50%;
    background-color: tomato;
    position: absolute;
    top: calc(40vh - 10px);
    left: calc(60vw - 40px);
    z-index: -1;
`;
const FirstPlanetDetail = styled.div`
    width: 20px;
    height: 20px;
    background-color: ;
    border-radius: 40%;

`;
const Home: React.FC = () => {
    const [reviews] = React.useState(() => {
        const rev = [];
        const firstMaleName = ['John', 'Andy', 'Michael', 'Steven', 'Dwight', 'Jim'];
        const firstFemaleName = ['Karen', 'Angela', 'Pam', 'Holly', 'Phyllis', 'Meredith', 'Erin', 'Jan'];
        const surName = ['Gonzales', 'Harper', 'Scott',]
        for (let i = 0; i < 3; i++) {
            rev.push({
                
            });
        }
    })


	return (<>
        <AppBar />
		<div style={{backgroundColor: 'hsl(221, 51%, 14%)'}}>			
            <FirstContainer>
            
                <LightText>
                    The voyage of your dreams starts now!
                </LightText>

                <LightText style={{textAlign: 'left', marginLeft: '2em'}}>
                    Ok then
                </LightText>

                {/* <BlueCircle />
                <FirstPlanet > 
                    <FirstPlanetDetail></FirstPlanetDetail>
                </FirstPlanet> */}                
            </FirstContainer>
            <SecondContainer>
            {[<div>First you select one of the available <Hl>planets</Hl></div>, 
            <div>Then, you choose one of the <Hl>spaceship</Hl> and <Hl>technology</Hl> plan</div>, 
            <div>Checkout time! You can pay using <Hl>credit card</Hl>, <Hl>online transfer</Hl> and <Hl>cryptocurrency</Hl>!</div>, 
            <div>Enjoy your outworldly vacation!</div>].map((e: any, i: number) => <ExplainBox style={{
                borderTop: `4px solid hsl(${i*125}, 91%, 54%)`
            }}>
                {e}
            </ExplainBox>)}
            </SecondContainer>
            <ThirdContainer>
                {
                    
                }
            </ThirdContainer>
		</div>
        </>
	);
};

export default Home;
