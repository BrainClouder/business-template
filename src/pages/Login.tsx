import React from 'react';
import styled from 'styled-components';

const isDesktop = window.innerWidth > 600;

const MainContainer = styled.div`
	background-image: url(https://picsum.photos/${window.innerWidth}/${window.innerHeight}?grayscale);
	background-color: hsl(249, 49%, 15%);
	position: absolute;
	height: 100vh;
	width: ${isDesktop ? '62vw' : '100vw'};
	right: 0;
	opacity: 0.8;
`;

const FormContainer = styled.div`
	background-color: hsl(221, 51%, 14%);
	box-shadow: 2px 2px 2px black;
	width: ${isDesktop ? '40vw' : '300px'};
	position: absolute;
	height: 100vh;
	left: 0;
`;

const Form = styled.div`
	background-color: hsl(221, 51%, 14%);
	padding: 1em;
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	align-items: center;
	position: absolute;
	border-radius: 20px;
	width: ${isDesktop ? '350px' : '300px'};
	left: ${isDesktop ? 'calc(20vw - 175px)' : '0'};
	top: 25vh;
`;

const InputText = styled.input`
	background-color: hsl(249, 49%, 85%);
	border-radius: 5px;
	text-align: center;
	font-size: 20px;
	color: hsl(221, 51%, 15%);
	width: 300px;
	box-shadow: inset 0 0 10px black;
	// border: 4px dashed hsl(221, 51%, 54%);
	height: 28px;
`;

const PrimaryButton = styled.button`
	background-color: hsl(221, 51%, 64%);
	padding: 10px 15px;
	border-radius: 10px;
    color: hsl(221, 51%, 98%);
    text-shadow: 1px 1px 2px hsl(221, 51%, 38%);
	font-size: 16px;
	font-family: Arial;
	letter-spacing: 3px;
	border: 2px solid hsl(221, 51%, 74%);
	box-shadow: 0 0 10px hsl(221, 51%, 8%);
	text-transform: uppercase;
	font-weight: 600;
	transition: 200ms;
	&:hover {
		background-color: hsl(221, 51%, 74%);
	}
`;

const SecondaryButton = styled.button`
	background-color: hsl(221, 51%, 14%);
	padding: 10px 15px;
	border-radius: 10px;
	color: hsl(221, 51%, 84%);
	font-size: 16px;
	font-family: Arial;
	letter-spacing: 3px;
	border: 2px solid hsl(221, 51%, 64%);
    box-shadow: 0 0 10px hsl(221, 51%, 8%);
    text-shadow: 1px 1px 2px hsl(221, 51%, 38%);
	text-transform: uppercase;
	font-weight: 400;
	transition: 200ms;
	&:hover {
		background-color: hsl(221, 51%, 44%);
	}
`;

const LabelText = styled.p`
	text-align: center;
	font-size: 18px;
	font-weight: 200;
	color: hsl(221, 51%, 95%);
`;

const ButtonContainer = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    width: 100%;
    margin: 0 auto;
`;

const InputContainer = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	margin: 8px;
	align-items: center;
`;

const Highlight = styled.span`font-weight: 700;`;

const SmallLabel = styled.p`
	font-size: 12px;
	font-weight: 200;
	margin-top: 5px;
	color: hsl(221, 51%, 75%);
`;

const Login: React.FC = () => {
    return (
        <div>
            <MainContainer />
            <FormContainer>
                <LabelText
                    style={{
                        color: 'hsl(221, 51%, 75%)',
                        fontSize: '32px',
                        letterSpacing: '2px',
                    }}
                >
                    Welcome back
				</LabelText>

                <Form>
                    <InputContainer>
                        <LabelText>Username or e-mail</LabelText>
                        <InputText type={'text'} />
                    </InputContainer>
                    <InputContainer>
                        <LabelText>Your password</LabelText>
                        <InputText type={'password'} />
                        <SmallLabel>Forgot your password?</SmallLabel>
                    </InputContainer>
                    <ButtonContainer>
                        <PrimaryButton>
                            enter
                    </PrimaryButton>
                        <SecondaryButton>
                            register
                    </SecondaryButton>
                    </ButtonContainer>
                </Form>
            </FormContainer>
        </div>
    );
};

export default Login;
