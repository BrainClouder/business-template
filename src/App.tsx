import React from 'react';
import { Switch, Route, Redirect } from 'react-router';
import Home from './pages/Home';
import Login from './pages/Login';

const App: React.FC = () => {
  return (
    <Switch>
      <Route path={`/home`} exact component={() => <Home/>} />
      <Route path={`/login`} exact component={() => <Login/>} />
      <Route path="" component={() => <Redirect to={"/home"} /> } />
    </Switch>
  );
}

export default App;
