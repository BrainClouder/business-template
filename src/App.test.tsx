import React from 'react';
// import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, withRouter } from 'react-router-dom';
import store from './store/reducer/main';
import Login from './pages/Login';
import Home from './pages/Home';

test('main app renders corretly (home page)', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Login page renders correctly', () => {
  const tree = renderer.create(<Login/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Home page renders correctly', () => {
  const tree = renderer.create(<>{withRouter(Home)}</>).toJSON();
  expect(tree).toMatchSnapshot();
});