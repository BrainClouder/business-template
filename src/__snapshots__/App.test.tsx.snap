// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `null`;

exports[`Login page renders correctly 1`] = `
<div>
  <div
    className="sc-fzpans loQKfm"
  />
  <div
    className="sc-fzplWN hnTcLr"
  >
    <p
      className="sc-fzqBZW ehHZWI"
      style={
        Object {
          "color": "hsl(221, 51%, 75%)",
          "fontSize": "32px",
          "letterSpacing": "2px",
        }
      }
    >
      Welcome back
    </p>
    <div
      className="sc-fznyAO hXymER"
    >
      <div
        className="sc-fzoyAV fERsLW"
      >
        <p
          className="sc-fzqBZW ehHZWI"
        >
          Username or e-mail
        </p>
        <input
          className="sc-fznKkj jhmTRc"
          type="text"
        />
      </div>
      <div
        className="sc-fzoyAV fERsLW"
      >
        <p
          className="sc-fzqBZW ehHZWI"
        >
          Your password
        </p>
        <input
          className="sc-fznKkj jhmTRc"
          type="password"
        />
        <p
          className="sc-fzoXzr clclza"
        >
          Forgot your password?
        </p>
      </div>
      <div
        className="sc-fzqNJr eSClwp"
      >
        <button
          className="sc-fznZeY gUAlQb"
        >
          enter
        </button>
        <button
          className="sc-fzokOt Bwapy"
        >
          register
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`main app renders corretly (home page) 1`] = `
Array [
  <div
    className="sc-AxjAm iIRRhm"
  >
    <div
      className="sc-AxirZ dXJXzm"
    >
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
        style={
          Object {
            "color": "hsl(221, 51%, 94%)",
            "fontSize": "18px",
            "fontWeight": 300,
            "textDecoration": "none",
            "textTransform": "capitalize",
          }
        }
      >
        login
      </a>
    </div>
    <div
      className="sc-AxirZ dXJXzm"
    >
      <a
        aria-current={null}
        href="/about us"
        onClick={[Function]}
        style={
          Object {
            "color": "hsl(221, 51%, 94%)",
            "fontSize": "18px",
            "fontWeight": 300,
            "textDecoration": "none",
            "textTransform": "capitalize",
          }
        }
      >
        about us
      </a>
    </div>
    <div
      className="sc-AxirZ dXJXzm"
    >
      <a
        aria-current={null}
        href="/pricing"
        onClick={[Function]}
        style={
          Object {
            "color": "hsl(221, 51%, 94%)",
            "fontSize": "18px",
            "fontWeight": 300,
            "textDecoration": "none",
            "textTransform": "capitalize",
          }
        }
      >
        pricing
      </a>
    </div>
    <div
      className="sc-AxirZ dXJXzm"
    >
      <a
        aria-current={null}
        href="/contact"
        onClick={[Function]}
        style={
          Object {
            "color": "hsl(221, 51%, 94%)",
            "fontSize": "18px",
            "fontWeight": 300,
            "textDecoration": "none",
            "textTransform": "capitalize",
          }
        }
      >
        contact
      </a>
    </div>
    <div
      className="sc-AxirZ dXJXzm"
    >
      <a
        aria-current={null}
        href="/workplace"
        onClick={[Function]}
        style={
          Object {
            "color": "hsl(221, 51%, 94%)",
            "fontSize": "18px",
            "fontWeight": 300,
            "textDecoration": "none",
            "textTransform": "capitalize",
          }
        }
      >
        workplace
      </a>
    </div>
  </div>,
  <div
    style={
      Object {
        "backgroundColor": "hsl(221, 51%, 14%)",
      }
    }
  >
    <div
      className="sc-AxiKw Ahors"
    >
      <p
        className="sc-AxheI isiLWF"
      >
        The voyage of your dreams starts now!
      </p>
      <p
        className="sc-AxheI isiLWF"
        style={
          Object {
            "marginLeft": "2em",
            "textAlign": "left",
          }
        }
      >
        Ok then
      </p>
    </div>
    <div
      className="sc-AxhCb BWqfa"
    >
      <div
        className="sc-AxgMl eKHRsh"
        style={
          Object {
            "borderTop": "4px solid hsl(0, 91%, 54%)",
          }
        }
      >
        <div>
          First you select one of the available 
          <span
            className="sc-Axmtr dFWOMB"
          >
            planets
          </span>
        </div>
      </div>
      <div
        className="sc-AxgMl eKHRsh"
        style={
          Object {
            "borderTop": "4px solid hsl(125, 91%, 54%)",
          }
        }
      >
        <div>
          Then, you choose one of the 
          <span
            className="sc-Axmtr dFWOMB"
          >
            spaceship
          </span>
           and 
          <span
            className="sc-Axmtr dFWOMB"
          >
            technology
          </span>
           plan
        </div>
      </div>
      <div
        className="sc-AxgMl eKHRsh"
        style={
          Object {
            "borderTop": "4px solid hsl(250, 91%, 54%)",
          }
        }
      >
        <div>
          Checkout time! You can pay using 
          <span
            className="sc-Axmtr dFWOMB"
          >
            credit card
          </span>
          , 
          <span
            className="sc-Axmtr dFWOMB"
          >
            online transfer
          </span>
           and 
          <span
            className="sc-Axmtr dFWOMB"
          >
            cryptocurrency
          </span>
          !
        </div>
      </div>
      <div
        className="sc-AxgMl eKHRsh"
        style={
          Object {
            "borderTop": "4px solid hsl(375, 91%, 54%)",
          }
        }
      >
        <div>
          Enjoy your outworldly vacation!
        </div>
      </div>
    </div>
    <div
      className="sc-AxhUy dNxuhj"
    />
  </div>,
]
`;
